_posts/2015-11-13-svg-sprites-angular.md

```diff
@@ -0,0 +1,167 @@
+
+So generating sprites became very popular way to overcome those limitations. With joining multiple icons/assets in to one big asset sheet, we download all related assets with one http request.
+Of course, there is some discussion about what makes sense to group in sprites, to download only needed assets and not something used rarely, but that discussion is out of the scope of this article.
+
+Ok, using sprites is nice but why svg sprites?
+
+Imagine, you have a set of icons that you group in sprite and you have to support multiple devices (as we usually do) - desktop, ios devices, mobile devices, tablets,... - with different screens. In that case, you can't use the same sprite if you wan to utilize screen specific advantages. For example, retina screens need assets double the size then other screens and serving bigger icons to screens that do not need them,
+would cause an overhead. That is bad for multiple reasons, most obvious one being downloading more data then needed and therefore increase of download time.
+
+Ok..so I create 2 sprite maps and serve them based on device type? Well, is not that simple, what if your icons change color based on context they are displayed in? Very quickly, you can end up with numerous amount of sprite files and complex logic to deal with them. At this point things can get out of hand very quickly.
+
+## There is a better way!
+
+Meet [svgs] (https://sl.wikipedia.org/wiki/Scalable_Vector_Graphics)! Svg stands for Scalable Vector Graphic. Main advantage of svgs is that they are, as name suggests, scalable vector graphics. That means they will be sharp on all type of screens and in all sizes - you need just one asset to achieve that. But it gets better, you can also change icon color, width, height (and much more!) through css properties. That means, that if you can have your repeatable icons in form of svgs, then you have to make only one sprite and it will serve any purpose.
+
+## Generate svg sprite
+
+Here, we will assume, that you already have assets in svg format. You can take all your svgs and convert them to symbols. You can use a gulp/grunt (or any other task) to do it. Or if you prefer to manage you icons with external tools, you can use something like [fontastic] (http://fontastic.me).
+
+So let's assume we have 3 svg icons.
+
+facebook.svg
+
+```
+<svg width="30" height="30" viewBox="0 0 30 30" xmlns="http://www.w3.org/2000/svg" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns"><title>social_facebook</title><desc>Created with Sketch.</desc><path d="M0 15c0 8.285 6.715 15 15 15s15-6.715 15-15c0-8.284-6.715-15-15-15s-15 6.716-15 15zm18.552-4.635h-2.254c-.268 0-.564.35-.564.819v1.629h2.819v2.321h-2.819v6.971h-2.662v-6.971h-2.414v-2.321h2.414v-1.366c0-1.959 1.36-3.553 3.226-3.553h2.254v2.471z" sketch:type="MSShapeGroup"/></svg>
```

Comment:
You can specify the syntax of the code block to apply nice colors (see the "Code" section https://guides.github.com/features/mastering-markdown/#examples)

```SVG
<svg width="30" height="30" viewBox="0 0 30 30" xmlns="http://www.w3.org/2000/svg" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns"><title>social_facebook</title><desc>Created with Sketch.</desc><path d="M0 15c0 8.285 6.715 15 15 15s15-6.715 15-15c0-8.284-6.715-15-15-15s-15 6.716-15 15zm18.552-4.635h-2.254c-.268 0-.564.35-.564.819v1.629h2.819v2.321h-2.819v6.971h-2.662v-6.971h-2.414v-2.321h2.414v-1.366c0-1.959 1.36-3.553 3.226-3.553h2.254v2.471z" sketch:type="MSShapeGroup"/></svg>
```